### PR TITLE
Saga deduplication tool: test fixup and better finding of connection strings

### PR DESCRIPTION
**Pull request by Scooletz**
Connects to https://github.com/Particular/PlatformDevelopment/issues/634

### CODE REVIEW

**Comment by adamralph** [comment]
> Connects to: Particular/PlatformDevelopment#634

I removed the colon (it broke Waffle's parsing :wink:)

**Comment by melkio** [comment]
As discussed with @Scooletz we can improve tool feedback with a link to documentation when users run it with missing/wrong parameters.
Adding a sort of help could be useful too.

**Comment by Scooletz** [comment]
Use only `connectionStrings` without a parameter.

**Comment by Scooletz** [comment]
Consider the fact that a customer can have 100 saga types. Maybe scanning should be introduced.

File: src/NServiceBus.AzureStoragePersistence.SagaDeduplicator.Tests/ProgramTests.cs [hunk 1]
```diff
@@ -32,7 +33,7 @@ public ProgramTests()
         [SetUp]
         public void SetUp()
         {
-            cloudTable = cloudTableClient.GetTableReference("Deduplicator" + Guid.NewGuid().ToString().Replace("-", ""));
+            cloudTable = cloudTableClient.GetTableReference(typeof(TwoInstanceSagaState).Name);
```
Comment (melkio) [inline src/NServiceBus.AzureStoragePersistence.SagaDeduplicator.Tests/ProgramTests.cs]: Why not directly "TwoInstanceSagaState" string?

File: src/NServiceBus.AzureStoragePersistence.SagaDeduplicator/Program.cs [hunk 2]
```diff
@@ -6,82 +6,158 @@
...
+            if (!string.IsNullOrWhiteSpace(settings?.ConnectionString))
+            {
+                connectionString = settings.ConnectionString;
+            }
+            else
             {
-                if (ConfigurationManager.ConnectionStrings.Count != 1)
+                // try to parse the last param
+                CloudStorageAccount account;
+                var possibleConnectionString = args.Last();
```
Comment (melkio) [inline src/NServiceBus.AzureStoragePersistence.SagaDeduplicator/Program.cs]: Does arguments order matter?

Comment (Scooletz) [inline src/NServiceBus.AzureStoragePersistence.SagaDeduplicator/Program.cs]: Nope. It finds first matching.

File: src/NServiceBus.AzureStoragePersistence.SagaDeduplicator.Tests/ProgramTests.cs (hunk 1 above)
Comment (Scooletz) [inline src/NServiceBus.AzureStoragePersistence.SagaDeduplicator.Tests/ProgramTests.cs]: It does not use the `AzureSagaPersister` but that's the way `AzureSagaPersister` obtains the name of the table.

**Pull request merged**